FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_plain_text/DemoTasks/PlaintextMQTTExample.c

```diff
@@ -451,35 +552,93 @@ static void prvCreateMQTTConnectionWithBroker( MQTTContext_t * pxMQTTContext,
 static void prvMQTTSubscribeToTopic( MQTTContext_t * pxMQTTContext )
 {
     MQTTStatus_t xResult;
-    MQTTSubscribeInfo_t xMQTTSubscription[ 1 ];
 
     /***
      * For readability, error handling in this function is restricted to the use of
      * asserts().
      ***/
 
     /* Some fields not used by this demo so start with everything at 0. */
-    ( void ) memset( ( void * ) &xMQTTSubscription, 0x00, sizeof( xMQTTSubscription ) );
+    ( void ) memset( ( void * ) pxGlobalSubscriptionList, 0x00, sizeof( pxGlobalSubscriptionList ) );
 
     /* Subscribe to the mqttexampleTOPIC topic filter. This example subscribes to
      * only one topic and uses QOS0. */
-    xMQTTSubscription[ 0 ].qos = MQTTQoS0;
-    xMQTTSubscription[ 0 ].pTopicFilter = mqttexampleTOPIC;
-    xMQTTSubscription[ 0 ].topicFilterLength = ( uint16_t ) strlen( mqttexampleTOPIC );
+    pxGlobalSubscriptionList[ 0 ].qos = MQTTQoS0;
+    pxGlobalSubscriptionList[ 0 ].pTopicFilter = mqttexampleTOPIC;
+    pxGlobalSubscriptionList[ 0 ].topicFilterLength = ( uint16_t ) strlen( mqttexampleTOPIC );
 
     /* Get a unique packet id. */
     usSubscribePacketIdentifier = MQTT_GetPacketId( pxMQTTContext );
 
     /* Send SUBSCRIBE packet. */
     xResult = MQTT_Subscribe( pxMQTTContext,
-                              xMQTTSubscription,
-                              sizeof( xMQTTSubscription ) / sizeof( MQTTSubscribeInfo_t ),
+                              pxGlobalSubscriptionList,
+                              sizeof( pxGlobalSubscriptionList ) / sizeof( MQTTSubscribeInfo_t ),
                               usSubscribePacketIdentifier );
 
     configASSERT( xResult == MQTTSuccess );
 }
 /*-----------------------------------------------------------*/
 
+static void prvUpdateSubAckStatus( MQTTPacketInfo_t * pxPacketInfo )
+{
+    MQTTStatus_t xResult = MQTTSuccess;
+    uint8_t * pPayload = NULL;
+    size_t pSize = 0;
+
+    xResult = MQTT_GetSubAckStatusCodes( pxPacketInfo, &pPayload, &pSize );
+
+    /* MQTT_GetSubAckStatusCodes always returns success if called with packet info
+     * from the event callback and non-NULL parameters. */
+    configASSERT( xResult == MQTTSuccess );
+
+    /* Demo only subscribes to one topic, so only one status code is returned. */
+    *pxGlobalSubAckStatus = pPayload[ 0 ];
```

Comment:
I think we should be consistent in accessing pointers. It looks weird to me that `pxGlobalSubAckStatus` (declared as an array) is accessed with `*`, but `pPayload` (declared as a pointer) is accessed with `[]`. I think accessing both with `[]` would be the most logical